### PR TITLE
merge ready mob-session with rebased master

### PR DESCRIPTION
**Pull request by pheenomenon**
For https://jira.corp.adobe.com/browse/NUI-1425 : Prepare mob session branches for merge
This PR is to merge into mob-session
- code cleanup and ensure unit tests pass
- rebase with master
- pending 
  - update pipeline dependency once open sourced (only will then the GitHub CI will pass)

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -28,20 +28,15 @@ const { AssetComputeWorkerPipeline } = require('./worker-pipeline');
...
 
     return actionWrapper(async function (params) {
     
-        // Debug: Use sleep to ensure tests have time to grab logs
-        // const { promisify } = require('util');
-        // const sleep = promisify(setTimeout);
-        // await sleep(2000);
-        console.log("## worker(renditionCallback) -> actionWrapper");
-        //*/
+        console.log("worker(renditionCallback) -> actionWrapper");
```
Comment (tmathern) [inline lib/api.js:39]: I think we can remove this if we keep the log from L49.

**Review by tmathern — APPROVED** [review]

File: lib/api.js (hunk 1 above)
Comment (jdelbick) [reply lib/api.js:39]: +1

**Review by jdelbick — CHANGES_REQUESTED** [review]
there are some more logs to remove:
https://github.com/adobe/asset-compute-sdk/blob/423ef651b8bc7c88ca7d2e0d431e4c95fa488235/lib/api.js#L75 (not very helpful, i'd remove)

https://github.com/adobe/asset-compute-sdk/blob/423ef651b8bc7c88ca7d2e0d431e4c95fa488235/lib/worker-pipeline.js#L77 <- manifest missing from log message. I would just remove this log & keep the log on line [80](https://github.com/adobe/asset-compute-sdk/blob/423ef651b8bc7c88ca7d2e0d431e4c95fa488235/lib/worker-pipeline.js#L80) instead

https://github.com/adobe/asset-compute-sdk/blob/423ef651b8bc7c88ca7d2e0d431e4c95fa488235/lib/worker-pipeline.js#L146 (<- remove log, leftover from debugging)

https://github.com/adobe/asset-compute-sdk/blob/423ef651b8bc7c88ca7d2e0d431e4c95fa488235/lib/worker-pipeline.js#L179-L180 (<- these contain credentials, rendition urls, source urls, etc. must be redacted)

**Comment by pheenomenon** [comment]
Any idea why the GitHub actions pipeline aren't triggering ?

**Comment by tmathern** [comment]
@pheenomenon Probably because the PR is targeting the mob-session branch. The Github Action are triggered only for main/master.

File: lib/worker-pipeline.js [hunk 2]
```diff
@@ -173,8 +174,8 @@ class AssetComputeWorkerPipeline {
         // normalize source and rendition
         this.normalizeInputOuput(input, output);
         // TODO: type check to make sure input/output are objects so we don't throw here if they are not?
-        debug(`Input for refinePlan: ${JSON.stringify(input, null, 1)}`);
-        debug(`Output for refinePlan: ${JSON.stringify(output, null, 1)}`);
+        AssetComputeLogUtils.log(input,'Input for refinePlan:');
+        AssetComputeLogUtils.log(output,'Output for refinePlan:');
```
Comment (jdelbick) [inline lib/worker-pipeline.js:178]: could you share what this output looks like? (I remember it was tricky to get right when I originally looked into this: https://github.com/adobe/asset-compute-sdk/pull/131/files#diff-78b14646706f0830f6321b56729ae22332f1afbf2d2782808adf7b6142deda8c)

ideally, for debugging purposes, I'd like to keep the whole nested json logged, just with the urls redacted:
```
Output for refinePlan: {
 "features": {
  "autoTags": {
   "image": false,
   "text": {
    "maximumWords": 3,
    "minimumConfidence": 0.7
   },
   "video": {
    "minimumActionConfidence": 0.7,
    "minimumObjectConfidence": 0.7
   }
  },
  "colorTags": false,
  "featureVector": false
 },
 "fmt": "machine-metadata-json",
 "pipeline": true,
 "target": {
  "maxPartSize": 104857600,
  "minPartSize": 10485760,
  "urls": [ ...REDACTED...]
 },
 "userData": {
  "assetComputeClient": {
   "index": 0,
   "length": 1
  },
  "path": "75d46622-de3e-478a-a3d2-30ca81f29582/machine-metadata-json/bulldog-5071407_319x179.png/rendition0.machine-metadata-json"
 },
 "type": "machine-metadata-json"
}
```
[old logs](https://adobeiort.splunkcloud.com/en-US/app/search/search?q=search%20index%3D%22runtime_prod_ext%22%20activation_id%3D72174fbeb2b54751974fbeb2b5575101%20%7C%20sort%20_time%20%7C%20fields%20%2B%20_time%2Clog_message%20%7C%20fields%20-%20_raw%20%7C%20outputtext&display.page.search.mode=smart&dispatch.sample_ratio=1&earliest=-7d%40h&latest=now&display.page.search.showFields=0&display.events.type=table&display.page.search.tab=statistics&display.general.type=statistics&sid=1626971142.341352)

Comment (pheenomenon) [reply lib/worker-pipeline.js:178]: I didn't get a chance to run e2e tests. I verified the logs in build/test.logs by changing the params.json for `pipeline-png-webp` test

```
{
	"pipeline": true,
	"fmt": "webp",
        "accessToken": "SOME_accessToken"
	"errorReason": "RenditionFormatUnsupported"
}
```

output:
```
2021-07-22T13:51:12.238234200Z Output for refinePlan: : { pipeline: true,
2021-07-22T13:51:12.238274100Z   fmt: 'webp',
2021-07-22T13:51:12.238294100Z   accessToken: '[...REDACTED...]',
2021-07-22T13:51:12.238308200Z   type: 'image/webp' }
```

Comment (tmathern) [reply lib/worker-pipeline.js:178]: @jdelbick @pheenomenon Normally the filenames should be redacted too.

**Review by jdelbick — APPROVED** [review]

**Pull request merged**